cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+		}
+		return func(p *libpod.Pod) bool {
+			ctrs, err := runtime.ContainersInPod(p)
+			if err != nil {
+				return false
+			}
+			status, err := getPodStatus(ctrs)
+			if err != nil {
+				return false
+			}
+			if strings.ToLower(status) == filterValue {
+				return true
+			}
+			return false
+		}, nil
+	}
+	return nil, errors.Errorf("%s is an invalid filter", filter)
+}
+
+// generate the template based on conditions given
+func genPodPsFormat(format string, quiet, cgroup, labels, names, ids, status bool) string {
+	if format != "" {
+		// "\t" from the command line is not being recognized as a tab
+		// replacing the string "\t" to a tab character if the user passes in "\t"
+		return strings.Replace(format, `\t`, "\t", -1)
+	}
+	if quiet {
+		return formats.IDString
+	}
+	format = "table {{.ID}}\t{{.Name}}\t{{.Status}}"
+	if names || ids || status {
+		format += "\t{{.ContainerInfo}}"
+	} else {
+		format += "\t{{.NumberOfContainers}}"
+	}
+	if labels {
+		format += "\t{{.Labels}}"
+	}
+	if cgroup {
+		format += "\t{{.Cgroup}}\t{{.UsePodCgroup}}"
+	}
+	return format
+}
+
+func podPsToGeneric(templParams []podPsTemplateParams, JSONParams []podPsJSONParams) (genericParams []interface{}) {
+	if len(templParams) > 0 {
+		for _, v := range templParams {
+			genericParams = append(genericParams, interface{}(v))
+		}
+		return
+	}
+	for _, v := range JSONParams {
+		genericParams = append(genericParams, interface{}(v))
+	}
+	return
+}
+
+// generate the accurate header based on template given
+func (p *podPsTemplateParams) podHeaderMap() map[string]string {
+	v := reflect.Indirect(reflect.ValueOf(p))
+	values := make(map[string]string)
+
+	for i := 0; i < v.NumField(); i++ {
+		key := v.Type().Field(i).Name
+		value := key
+		if value == "ID" {
+			value = "Pod" + value
+		}
+		values[key] = strings.ToUpper(splitCamelCase(value))
+	}
+	return values
+}
+
+func sortPodPsOutput(sortBy string, psOutput podPsSorted) (podPsSorted, error) {
+	switch sortBy {
+	case "id":
+		sort.Sort(podPsSortedId{psOutput})
+	case "name":
+		sort.Sort(podPsSortedName{psOutput})
+	case "number":
+		sort.Sort(podPsSortedNumber{psOutput})
+	case "status":
+		sort.Sort(podPsSortedStatus{psOutput})
+	default:
+		return nil, errors.Errorf("invalid option for --sort, options are: id, names, or number")
+	}
+	return psOutput, nil
+}
+
+// getPodTemplateOutput returns the modified container information
+func getPodTemplateOutput(psParams []podPsJSONParams, opts podPsOptions) ([]podPsTemplateParams, error) {
+	var (
+		psOutput []podPsTemplateParams
+	)
+
+	for _, psParam := range psParams {
+		labels := formatLabels(psParam.Labels)
+		podID := psParam.ID
+		var ctrStr string
+
+		if !opts.NoTrunc {
+			podID = shortID(psParam.ID)
+			if opts.IdsOfContainers {
+				for i := 0; i < len(psParam.CtrsInfo); i++ {
+					psParam.CtrsInfo[i].Id = shortID(psParam.CtrsInfo[i].Id)
+				}
+			}
+		}
+		for _, ctrInfo := range psParam.CtrsInfo {
```

Comment:
We should pick a max number that we will iterate through, and if we have more containers, append a '...' after. I can see pods with 50-100 containers, and we don't want all of them on one line

Reply:
How does 10 unless no-trunc sound?

Reply:
SGTM